hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/interceptor/AbstractInterceptor.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.bytecode.enhance.spi.interceptor;
+
+import org.hibernate.engine.spi.PersistentAttributeInterceptor;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+
+/**
+ * @author Steve Ebersole
+ */
+public abstract class AbstractInterceptor implements PersistentAttributeInterceptor, SessionAssociableInterceptor {
```

Comment:
just a minor, `PersistentAttributeInterceptor` can be removed, `SessionAssociableInterceptor` already extends `PersistentAttributeInterceptor`

Reply:
Also, ```AbstractSessionAssociableInterceptor``` might be a better name.

Reply:
Should this class implement ```Serializable```?